jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/RegexPathSpec.java

```diff
@@ -140,7 +140,7 @@ public String getPathMatch(String path)
         Matcher matcher = getMatcher(path);
         if (matcher.matches())
         {
-            if (matcher.groupCount() >= 1)
+            if (_group == PathSpecGroup.PREFIX_GLOB && matcher.groupCount() >= 1)
```

Comment:
What is the reason for this change? 
This change is not present in 10 & 11.

Reply:
@lachlan-roberts my intention is to move this change forward into 10 & 11.
I believe it is needed to make the methods `getPathMatch` and `getPathInfo` symmetric.

Specifically I would expect either `pathMatch==/test && pathInfo==/info` or `pathMatch==/test/info && pathInfo==null`, but never `pathMatch==/test && pathInfo==null`, since that loses information.  Currently we get the later behavior!

So I have added the following test.

```java
    @Test
public void testPathInfo()
    {
        RegexPathSpec spec = new RegexPathSpec("^/test(/.*)?$");
        assertTrue(spec.matches("/test/info"));
        assertThat(spec.getPathMatch("/test/info"), equalTo("/test"));
        assertThat(spec.getPathInfo("/test/info"), equalTo("/info"));

        spec = new RegexPathSpec("^/[Tt]est(/.*)?$");
        assertTrue(spec.matches("/test/info"));
        assertThat(spec.getPathMatch("/test/info"), equalTo("/test/info"));
        assertThat(spec.getPathInfo("/test/info"), nullValue());
    }
```

Note that this issue was not apparent in 10 & 11, because they do not use these methods to set the servletPath and pathInfo, but rather work it out differently (I'll check exactly why when I port this change forward).